vars/createUploadTestReportManifest.groovy

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to run the report workflow to generate test-report manifest.
+ @param Map args = [:] args A map of the following parameters
+ @param args.testManifest <required> - The path of the test manifest used
+ @param args.buildManifest <required> - The path of the build manifest of OpenSearch artifacts.
+ @param args.dashboardsBuildManifest <optional> - The path of the build manifest of OpenSearch Dashboards
+ @param args.testRunID <required> - Test run id of the test workflow being reported.
+ @param args.testType <required> - Type of the test workflow being reported.
+ @param args.componentName <Optional> - Components that workflow runs on.
+ */
+
+def call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+
+    def testRunID = args.testRunID;
```

Comment:
Do you want to add parameterCheck function to get all the required params? Similar to https://github.com/opensearch-project/opensearch-build-libraries/blob/99c487d8be8540252bd73818a89d311711144b3e/vars/publishToNpm.groovy#L33-L47

Reply:
I believe I'm asserting and verifying those required parameter in the libtester to make sure they are not null.

Reply:
That won't verify on actual jenkins end. Lib tester is only for testing the library not actual checks.

Reply:
Oh I see your point here. Do you think it's that necessary? We would be passing these required parameters for sure in the integ test jenkinsfile instead of passing by ourself manually. Even the worst case that some of those parameters are not present, i don't think we want to fail the testing because of this reporting issue and then we have to run the tests again. The only error would only appear on the test-report manifest if that's the case.

Reply:
Its the opposite. This will only fail when we try to create the report and not the entire workflow. From library's perspective you can add what args are hard core required. I know we didn't do that before but this helps is fail fast and debug easier than figuring out in python code what went wrong.

Reply:
We can have an offline discussion on this. I may not get your points here since I think it might be not that helpful in this case.

Reply:
Had an offline discussion with @gaiksaya, since we are on pace to make all library with parameter checks, adding checks here as well. Please help review as the checks are added now. If the required parameter is null or empty, we won't fail the job but skip the rest of the workflow.